cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -12,22 +12,46 @@ WORKER_NODE_ROOT="${CLUSTER_NAME}-worker"
 
 OPERATOR_GIT_HASH=8d3c30de8ec5a9a0c9eeb84ea0aa16ba2395cd68  # release-4.4
 
-# not using kubectl wait since with the sriov operator the pods get restarted a couple of times and this is
-# more reliable
-function wait_pods_ready {
-    while [ -n "$(_kubectl get pods --all-namespaces -o'custom-columns=status:status.containerStatuses[*].ready,metadata:metadata.name' --no-headers | grep false)" ]; do
-        echo "Waiting for all pods to become ready ..."
-        _kubectl get pods --all-namespaces -o'custom-columns=status:status.containerStatuses[*].ready,metadata:metadata.name' --no-headers
-        sleep 10
-    done
+function wait_for {
+  retries=100
+  while [[ $retries -ge 0 ]]; do
+    if _kubectl get $@; then
+      break
+    fi
+    sleep 6
+    ((retries--))
+  done
+}
+
+function wait_for_daemonset {
+  retries=100
+  while [[ $retries -ge 0 ]]; do
+    ready=$(_kubectl -n $1 get daemonset $2 -o jsonpath='{.status.numberReady}')
```

Comment:
if both fail, then the condition on line 31 will be true
because the strings would match

Reply:
Good catch, thanks. For now I settled with simple check whether the returned value is not empty, but we may need to use one of the mechanisms you've suggested if the provider will be unstable.

Reply:
i think now its good enough, i dont see other issues since you already have a retry + non empty check, am i miss something?

Reply:
Well the only remaining issue is that the lane is still extremely unstable.